docs/4_0_0_apis.md

```diff
@@ -4,59 +4,67 @@ A handful of APIs are available to make patch development easier and more effici
+    val superClassOfReturnType = classBy(match().originalMethod.returnType).superclass
 }
 ```
 
-#### 🔍 `classBy(Predicate)`
+#### 👹 `mutableClassBy(ClassDef)`
 
-The `classBy(Predicate)` function is an alternative to finding and creating mutable classes by a predicate.
-It automatically proxies the class definition, making it mutable.
+By default, the classes are immutable and they cannot be modified.
+To make a class mutable use the `mutableClassBy(ClassDef)` function.
+Accessing the property will replace the original class definition with the mutable copy,
+thus allowing you to make changes to the class. Subsequent accesses will return the same mutable copy.
 
 ```kt
 execute {
-    // Alternative to proxy(classes.find { it.name == "Lcom/example/MyClass;" })?.classDef
-    val classDef = classBy { it.name == "Lcom/example/MyClass;" }?.classDef
+    // Find a class by the return type of a fingerprint
+    val superClassOfReturnType = classBy(match().originalMethod.returnType).superclass
+
+    val mutableClass = mutableClassBy(superClassOfReturnType)
+    mutableClass.methods.add(Method())
 }
 ```
 
 #### 🏃‍ `navigate(Method).at(index)`
 
-The `navigate(Method)` function allows you to navigate method calls recursively by index.
+The `navigate(Method)` function allows navigating method calls by index,
+and provides an easier way to parse the method call classes in code. 
 
 ```kt
 execute {
-    // Sequentially navigate to the instructions at index 1 within 'someMethod'.
-    val method = navigate(someMethod).to(1).original() // original() returns the original immutable method.
+    // Navigate to the method at index 5 within 'someMethod'.
+    // original() returns the original immutable method.
+    val original = navigate(someMethod).to(5).original()
     
-    // Further navigate to the second occurrence where the instruction's opcode is 'INVOKEVIRTUAL'.
+    // Further navigate to the second occurrence of the opcode 'INVOKE_VIRTUAL'.
     // stop() returns the mutable copy of the method.
-    val method = navigate(someMethod).to(2) { instruction -> instruction.opcode == Opcode.INVOKEVIRTUAL }.stop()
-    
-    // Alternatively, to stop(), you can delegate the method to a variable.
-    val method by navigate(someMethod).to(1)
+    val mutable = navigate(someMethod).to(2) { 
+        instruction -> instruction.opcode == Opcode.INVOKE_VIRTUAL
+    }.stop()
     
-    // You can chain multiple calls to at() to navigate deeper into the method.
-    val method by navigate(someMethod).to(1).to(2, 3, 4).to(5)
+    // You can chain multiple to() calls together navigate multiple calls across different methods and classes.
+    //
+    // Navigate to:
+    // A. the method of the 5th instruction
+    // B. the method of the 10th instruction in method A
+    // C. the method of 2nd instruction of method B
+    val mutableDeep = navigate(someMethod).to(5, 10, 2).stop() // Mutable method Method C
```

Comment:
Perhaps we can simplify this api by replacing recursion with iteration.

This would spare us the odd api "stop".

Instead we could have 

fun to(vararg indices: Int, mutable: boolean)

Chaining can be allowed with another overload:


fun Method.to(vararg indices: Int, mutable: boolean)

this way it will work with:

navigate(someMethod).to(1,2,3).to(4, mutable=true) 

We could maybe even get rid of more abstraction and just provide:

navigate(someMethod, 1,2,3, mutable)

and maybe even normalize the name for example with

method(from=someMethod, at=1, mutable=true) (or getMethod etc)